cli/flox-rust-sdk/src/flox.rs

```diff
@@ -54,10 +54,20 @@ pub struct Flox {
     pub floxhub_token: Option<FloxhubToken>,
 
     pub catalog_client: Option<catalog::Client>,
+
+    /// Feature flags
+    pub features: Features,
 }
 
 impl Flox {}
 
+#[derive(Debug, Clone, Deserialize, Serialize, Default)]
+pub struct Features {
```

Comment:
suggestion nonblocking: the `Features` struct in the flox crate has a `services` field, a `search_strategy` field which we'll drop, and a `use_catalog` field that we use extensively in flox-rust-sdk, although indirectly via checking if we have a catalog client. Can we just drop the struct in the flox crate? Having two `Features` structs seems confusing - does it gain us anything?

Reply:
I'd prefer we drop the one in the `flox` crate

Reply:
I'm not sure I follow..

Are you suggesting that we remove `search_strategy` and `use_catalog` now, ahead of any work to remove pkgdb?

Doesn't removing the outer `flox::config::Feature` struct affect our ability to flag with environment variables?

Reply:
I would pull this struct here into the config e.g.

```
Config {
	features: flox_rust_sdk::Features
}
```

and either move the existing features into this one as well or drop them (though i think we shouldnt drop use_catalog just yet)

or extend the `flox::config::features` cli specific features with library features:

```
Features {
	...
	#[serde(flatten)]
	lib: flox_rust_sdk::Features
}

```


-------

the config was originally defined in `flox` as it foremost changed the cli layer which then configures the library by constructing a `Flox` object where some attributes may be derived from the config.
Again separating user interface from implementation.
For features the idea was more to provide any feature at all times in the cli, but provide separate methods or arguments to choose different library behavior from the cli layer. In how far that's worth it in cases like this where the changed behaviour is all the way into `CoreEnvironment` is debatable.
I do see an argument to make features part of the library/Flox context, though there may still be cli only features that just dont make sense on the lib layer, thus the second suggestion above, which would allow for both kind of toggles...

Reply:
From sync: move the whole struct down as-is. I'll do it in a follow-up PR because this one already has a lot of churn.

Reply:
@ysndr I do like doing it with `#[serde(flatten)]` a good bit more than just having 2 totally separate structs. I think if we encounter a CLI only feature we can add back in a 2nd struct but for now it's simpler to just have one.